packages/components/src/ui/flex/use-flex.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { useContextSystem } from '@wp-g2/context';
+import { css, cx, ui, useResponsiveValue } from '@wp-g2/styles';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from './flex-styles';
+
+/**
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<import('./types').FlexProps, 'div'>} props
+ */
+export function useFlex( props ) {
+	const {
+		align = 'center',
+		className,
+		direction: directionProp = 'row',
+		expanded = true,
+		gap = 2,
+		justify = 'space-between',
+		wrap = false,
+		...otherProps
+	} = useContextSystem( props, 'Flex' );
+
+	const directionAsArray = Array.isArray( directionProp )
+		? directionProp
+		: [ directionProp ];
+	const direction = useResponsiveValue( directionAsArray );
+
+	const isColumn =
+		typeof direction === 'string' && !! direction.includes( 'column' );
+	const isReverse =
+		typeof direction === 'string' && direction.includes( 'reverse' );
+
+	const classes = useMemo( () => {
+		const sx = {};
+
+		sx.Base = css( {
```

Comment:
Oo! `Flex` in G2 was recently enhanced to better accommodate `wrap` layouts. (https://github.com/ItsJonQ/g2/pull/252)

For `Flex`, the changes are scoped to `useFlex` and `Flex.styles`
https://github.com/ItsJonQ/g2/blob/main/packages/components/src/Flex/useFlex.js#L65
https://github.com/ItsJonQ/g2/blob/main/packages/components/src/Flex/Flex.styles.js#L31